### ISSUE TITLE
File/Folder-names in import statements are not case-sensitive

### ISSUE THREAD

**Issue by etecture-plaube** [palantir/tslint#3291]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.1
- __Running TSLint via__: CLI

#### TypeScript code being linted
index.ts
```ts
import FooBar from './foobar';
```
FooBar.ts
```ts
export default console.log('FooBar');
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:latest"
  ],
  "rules": {
    "no-require-imports": false,
    "no-var-requires": false,
    "quotemark": [true, "single", "jsx-double"],
    "no-console": false,
    "no-submodule-imports": false,
    "space-within-parens": [false, 1],
    "whitespace": {
      "options": [
        "check-branch",
        "check-decl",
        "check-module",
        "check-operator",
        "check-separator",
        "check-type",
        "check-typecast",
        "check-type-operator",
        "check-rest-spread",
        "check-preblock"
      ]
    }
  }
}
```

#### Actual behavior
No error, no warning.

#### Expected behavior
Error: File './foobar.ts' not found.

#### Info
The describled behaviour is on MAC OS, where filesystem paths are case-insensitive. 
On Linux or Windows we get a TS (not TSLINT) Error: TS2307 Cannot find module './foobar.ts'.

**Comment by ajafff** [comment]
So this is not about any linting rule that currently exists, right?
On case insensitive file systems that's expected behavior, one could even argue it's a feature.

After all TSLint does not try to resolve your imports. It defers that to TypeScript, which knows how to do that, because you specify it in your tsconfig.json. That's why you should rather open this issue in the TypeScript repository. Or even better: look into using the `--forceConsistentCasingInFileNames` compiler option and search their issue tracker for similar issues, because there have been plenty of them. I just cannot find them right now.

**Comment by etecture-plaube** [comment]
Thank you very much. `--forceConsistentCasingInFileNames` is exactly what i was looking for.

**Comment by alexicum** [comment]
[Rule for case sensitive imports](https://github.com/palantir/tslint/issues/2503)
